test/functional/test_framework/test_framework.py

```diff
@@ -627,19 +625,28 @@ def connect_nodes(self, a, b, *, peer_advertises_v2=None, wait_for_connect: bool
         if not wait_for_connect:
             return
 
+        # Use subversion as peer id
+        from_connection_subver = from_connection.getnetworkinfo()['subversion']
+        to_connection_subver = to_connection.getnetworkinfo()['subversion']
```

Comment:
Looking at the sample output of `subversion` - `"subversion": "\/Satoshi:25.1.0\/"`, it doesn't seem unique enough because multiple nodes can be running the same version. Won't this cause issues in `find_conn` later?

https://chainquery.com/bitcoin-cli/getnetworkinfo

Reply:
> Looking at the sample output of `subversion` - `"subversion": "\/Satoshi:25.1.0\/"`, it doesn't seem unique enough because multiple nodes can be running the same version. Won't this cause issues in `find_conn` later?

The test framework appends the node number to the user agent string. See [test_node](https://github.com/bitcoin/bitcoin/blob/2f53f2273da020d7fabd7c65a1bc7e69a31249b2/test/functional/test_framework/test_node.py#L109).

Reply:
Worth adding a note to the comment?

```python
        # Use subversion as peer id -- test nodes have their number appended to the UA
```

Reply:
Worth adding a note to the comment?

```python
        # Use subversion as peer id. Test nodes have their node number appended to the UA
```

Reply:
> Worth adding a note to the comment?
> 
> ```python
>         # Use subversion as peer id. Test nodes have their node number appended to the UA
> ```

sure. Pushed.